libcloudforensics/providers/gcp/internal/storagetransfer.py

```diff
@@ -0,0 +1,160 @@
+        https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferOperations/list  # pylint: disable=line-too-long
+    Raises:
+      TransferCreationError: If the transfer couldn't be created.
+      TransferExecutionError: If the transfer couldn't be run.
+    """
+    aws_creds = account.AWSAccount(zone).session.get_credentials()
+    s3_bucket, s3_path = gcp_storage.SplitStoragePath(s3_path)
+    gcs_bucket, gcs_path = gcp_storage.SplitStoragePath(gcs_path)
+    if not gcs_path.endswith('/'):
+      gcs_path = gcs_path + '/'
+    today = datetime.datetime.now()
+    transfer_job_body = {
+        'projectId': self.project_id,
+        'description': 'created_by_cfu',
+        'transferSpec': {
+            'objectConditions': {
+                'includePrefixes': [s3_path]
+            },
+            'awsS3DataSource': {
+                'bucketName': s3_bucket,
+                'awsAccessKey': {
+                    'accessKeyId': aws_creds.access_key,
+                    'secretAccessKey': aws_creds.secret_key
+                }
+            },
+            'gcsDataSink': {
+                'bucketName': gcs_bucket, 'path': gcs_path
+            }
+        },
+        'schedule': {
+            'scheduleStartDate': {
+                'year': today.year, 'month': today.month, 'day': today.day
+            },
+            'scheduleEndDate': {
+                'year': today.year, 'month': today.month, 'day': today.day
+            },
+            'endTimeOfDay': {}
+        },
+        'status': 'ENABLED'
+    }
+    logger.info('Creating transfer job')
+    gcst_jobs = self.GcstApi().transferJobs()
+    create_request = gcst_jobs.create(body=transfer_job_body)
+    transfer_job = create_request.execute()
+    logger.info('Job created: {0:s}'.format(str(transfer_job)))
+    job_name = transfer_job.get('name', 'Error: name not found')
+    if 'Error' in job_name:
+      raise errors.TransferCreationError(
+          'Could not create transfer. Job output: {0:s}'.format(
+              str(transfer_job)),
+          __name__)
+    logger.info('Job created: {0:s}'.format(job_name))
+    gcst_transfers = self.GcstApi().transferOperations()
+    filter_string = ('{{"projectId": "{0:s}", "jobNames": ["{1:s}"]}}').format(
+        self.project_id, job_name)
+    status = {}
+    while 'operations' not in status:
+      time.sleep(5)
+      status = gcst_transfers.list(
+          name='transferOperations', filter=filter_string).execute()
+      logger.info('Waiting for transfer to start...')
+    logger.info('Job status: {0:s}'.format(str(status)))
+    while status['operations'][0].get('metadata', {}).get('status',
+                                                          '') == 'IN_PROGRESS':
+      time.sleep(5)
+      status = gcst_transfers.list(
+          name='transferOperations', filter=filter_string).execute()
+      logger.info('Waiting to finish...')
+    for operation in status.get('operations', []):
```

Comment:
why not status['operations'][0] here, shouldn't it be one operation that is returned when using the filter_string? and if it can be more than one operation shouldn't a loop be used above when checking status rather than using status['operations'][0]?